PSICompile/ILCodeGen.cs

```diff
@@ -121,7 +136,7 @@ public class ILCodeGen : Visitor {
    public override void Visit (NUnary u) {
       u.Expr.Accept (this);
       string op = u.Op.Kind.ToString ().ToLower ();
-      op = op switch { "sub" => "neg", _ => op };
+      op = op switch { "sub" => "neg", "not" => "ldc.i4.1 xor", _ => op };

```

Comment:
`WriteLn (not 2)` should print **-3**, which is the integer value you get when you invert all the bits in the binary representation of 2. Your code will work fine for boolean (though `ldc.i4.0 ceq` is more robust - it will treat all non-zero values as true).

Reply:
Ok. Understood. Updated to ldc.4.0 ceq.